provider/management/commands/delete_expired_grant_tokens.py

```diff
@@ -0,0 +1,53 @@
+"""
+Command to delete expired OAuth2 grant tokens from the oauth2_grant table.
+"""
+
+import logging
+from datetime import datetime, timedelta
+from django.core.management.base import BaseCommand
+from django.db import transaction
+from provider.oauth2.models import Grant
+
+log = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    """
+    Example usage: ./manage.py lms --settings=devstack delete_expired_grant_tokens.py
+    """
+    help = 'Deletes all expired OAuth2 grant tokens (in chunks).'
+
+    # Maximum number of expired tokens to delete in a single transaction.
+    CHUNK_SIZE = 10000
+
+    def handle(self, *args, **options):
+        """
+        Deletes oauth2_grant tokens expired for days_back days, chunking the deletes to avoid long table/row locks.
+        """
+        table_name = 'oauth2_grant'
+        delete_date = datetime.now()
+        where_clause = "WHERE expires < '{}'".format(delete_date.strftime('%Y-%m-%d %H:%m:%S'))
+        # The "as id" below fools Django raw query into thinking the primary key is being queried.
+        total_expired_tokens = Grant.objects.raw(
+            'SELECT COUNT(*) as id FROM {} {}'.format(table_name, where_clause)
+        )[0].id
+        log.info("STARTED: Deleting %s expired grant tokens older than %s.", total_expired_tokens, delete_date)
+
+        total_deletions = 0
+        while total_deletions < total_expired_tokens:
+            deletions_now = min(self.CHUNK_SIZE, total_expired_tokens - total_deletions)
+            log.info("Deleting {} expired tokens...".format(deletions_now))
+            with transaction.atomic():
+                delete_sql = 'DELETE FROM {} WHERE id IN (SELECT id FROM {} {} LIMIT {})'.format(
+                    table_name, table_name, where_clause, deletions_now
+                )
```

Comment:
Ok, redirect_uri is a safe selector.  I didn't realize oidc actually pushed out the expires date.  Let's go with the subselect, but keep in mind that we'll need to build into our time calculation the few seconds of selects on each one.

@doctoryes My only request would be an argument for --sleep-between that lets you sleep for 3 seconds between chunks (and making CHUNK_SIZE an optional argument so we can tune it).

We frequently find in prod that we need to do slow deletes (3000 or 5000 rows at a time, 5 seconds between deletes) to avoid causing disk queueing at the EBS level that backs up other reads/writes.